pandas/_config/config.py

```diff
@@ -461,17 +461,20 @@ def register_option(key: str, defval: object, doc="", validator=None, cb=None):
             raise ValueError(f"{k} is a python keyword")
 
     cursor = _global_config
-    msg = "Path prefix to option '{option}' is already an option"
 
     for i, p in enumerate(path[:-1]):
         if not isinstance(cursor, dict):
-            raise OptionError(msg.format(option=".".join(path[:i])))
+            raise OptionError(
+                f"Path prefix to option '{'.'.join(path[:i])}' is already an option"
```

Comment:
I think @jbrockmendel prefers to define expressions like this in a variable first, instead of using them directly in the f-string.

Reply:
yes i do.  thanks for keeping an eye out

Reply:
my preference would be to leave the .format syntax when message used more than once to help ensure consistency of error messages.

Reply:
Reverted to the original .format syntax, but will keep the earlier comments in mind for the future!